packages/scandipwa/src/component/ProductBundleItems/ProductBundleItems.container.js

```diff
@@ -73,42 +78,94 @@ export class ProductBundleItemsContainer extends ProductCustomizableOptionsConta
         this.setState({ isLoading: false });
     }
 
-    getOptionPrice(item, selectedValues) {
-        const { option_id } = item;
-        let price = 0;
-        let priceExclTax = 0;
+    optionPriceMap = {
+        [PRICE_TYPE_DYNAMIC]: this.getDynamicOptionPrice.bind(this),
+        [PRICE_TYPE_FIXED]: this.getFixedOptionPrice.bind(this),
+        [PRICE_TYPE_PERCENT]: this.getPercentOptionPrice.bind(this)
+    };
+
+    getDynamicOptionPrice(product, quantity) {
+        const {
+            price_range: {
+                minimum_price: {
+                    regular_price: {
+                        value: itemPrice = 0
+                    } = {},
+                    final_price: {
+                        value: finalItemPrice = 0
+                    } = {},
+                    final_price_excl_tax: {
+                        value: finalItemPriceExclTax = 0
+                    } = {}
+                } = {}
+            } = {}
+        } = product;
+
+        const optionInitialPrice = itemPrice * quantity;
+        const optionPrice = finalItemPrice * quantity;
+        const optionPriceExclTax = finalItemPriceExclTax * quantity;
+        return { optionPrice, optionPriceExclTax, optionInitialPrice };
+    }
+
+    getPercentOptionPrice(product, quantity, optionPricePercent) {
+        const {
+            price_range: {
+                minimum_price: {
+                    default_price: { value: defaultPrice } = {},
+                    default_final_price: { value: defaultFinalPrice } = {},
+                    default_final_price_excl_tax: { value: defaultFinalPriceExclTax } = {}
+                }
+            }
+        } = this.props;
+
+        const optionPriceMultiplier = optionPricePercent / ONE_HUNDRED_PERCENT;
+
+        const optionInitialPrice = defaultPrice * optionPriceMultiplier * quantity;
+        const optionPrice = (defaultFinalPrice * optionPriceMultiplier * quantity);
+        const optionPriceExclTax = (defaultFinalPriceExclTax * optionPriceMultiplier * quantity);
+        return { optionPrice, optionPriceExclTax, optionInitialPrice };
+    }
+
+    getFixedOptionPrice(product, quantity, optionPriceFixed) {
+        const optionInitialPrice = optionPriceFixed * quantity;
+        const optionPrice = optionPriceFixed * quantity;
+        const optionPriceExclTax = optionPriceFixed * quantity;
+        return { optionPrice, optionPriceExclTax, optionInitialPrice };
```

Comment:
Also this. Maybe, one comment will be enough if the meanings are the same for all of the cases.